### DISCUSSION TITLE
How to depend on swift runtime support?

### DISCUSSION THREAD

**Discussion by Cyberbeni** [Homebrew/discussions#68] [Everyday usage] [answered]
Starting with macOS 10.14.4, it is integrated into the OS, otherwise you need to install the package from here: https://support.apple.com/kb/DL1998

How would you define this dependency in a formula?

**Comment by SMillerDev** [comment]
You could do `uses_from_macos "swift"`

**Reply by Cyberbeni** [reply]
Doesn't that just mean that if the `swift` command in shell is not available then fail?

**Reply by SMillerDev** [reply]
No, it means that it expects the equivalent of the swift formula to be provided by macOS.

**Reply by Cyberbeni** [reply]
Yeah, the swift formula provides the build tools. What I'm asking would be useful when my formula is about a prebuilt cli tool that requires the swift 5 runtime, which is included in macOS 10.14.4+ and can be either embedded to support earlier versions or downloaded from the link I posted above. And since the primary use case is usually newer OS versions, it doesn't make sense embedding it over requiring the OS to have the runtime.

**Reply by SMillerDev** [reply]
There's no real way to depend on system components. The best you can do is require mojave in this case.

**Reply by Cyberbeni** [reply]
I found this cask for it: https://github.com/thii/homebrew-swift-runtime/blob/master/Casks/swift-runtime.rb
Can I do  `depends_on cask: "thii/swift-runtime/swift-runtime" if MacOS.version < "10.14.4"`

**Reply by SMillerDev** [reply]
Only if it's in a personal tap. Because formulae in homebrew/core can't depend on taps.

**Comment by Cyberbeni** [comment, marked as answer]
`depends_on cask: "thii/swift-runtime/swift-runtime" if MacOS.version < "10.14.4"`
